### DISCUSSION TITLE
[JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.

### DISCUSSION THREAD

**Discussion by subashiniramesh17** [davidzajac1/zillacode#3] [General]
I have done all the steps mentioned in the github and installed docker , now when i try to run the code in the local host page , im getting the below error.

[JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.
Please suggest ways to resolve this issue , Im not able to compile any of my code

**Comment by davidzajac1** [comment]
Thanks @subashiniramesh17 can you comment with the following info
- What kind of machine you're using (Windows, Mac M2, etc.)
- What language this issue is showing for (PySpark or Scala Spark)
- A screenshot of the error

**Reply by subashiniramesh17** [reply]
Thankyou for your response . Below are the required info ,
1. Im using a Windows machine
2.This issue is showing for Pyspark
3. Screenshot attached below
![image](https://github.com/user-attachments/assets/736b420e-fc65-44bf-b611-a3fe164cf05b)

**Reply by davidzajac1** [reply]
Thanks! The issue is that the zillacode-spark container isn't finding JAVA_HOME in the path.

While everything is still running can you try running `docker-compose up -d --no-deps --build spark` in a different terminal?

If that doesn't work could you open Docker Desktop and paste the logs from the zillacode-spark container?

**Comment by davidzajac1** [comment]
@subashiniramesh17 this was fixed with this PR: https://github.com/davidzajac1/zillacode/pull/5

If you re-clone the repo and run `docker-compose up` it should work now.
